File: Haskell/prob7.hs
--isPrime taken from stackoverflow, but I finally understand it.

isPrime n = go 2
	where
		go d
			| d*d > n = True
			| n `mod` d == 0 = False
			| otherwise = go (d+1)

primes = 2:filter isPrime [3,5..]

main = print (primes!!10000)

File: Python/Prob7.py
#works. isPrime taken from Prob3
primes = []
i = 1
p = 0
done = False

def isPrime(num, d):
	if d*d > num: return True
	elif num%d==0: return False
	else: return isPrime(num, d+1)

while p<=10001:
	if isPrime(i, 2):
		primes.append(i)
		p+=1
	i+=1

print max(primes)